controllers/keda/util/predicate.go

```diff
@@ -89,3 +92,14 @@ func (PausedPredicate) Update(e event.UpdateEvent) bool {
 
 	return newPausedValue != oldPausedValue
 }
+
+type HPAUpdatePredicate struct {
+	predicate.Funcs
+}
+
+func (HPAUpdatePredicate) Update(e event.UpdateEvent) bool {
+	newObj := e.ObjectNew.(*autoscalingv2.HorizontalPodAutoscaler)
+	oldObj := e.ObjectOld.(*autoscalingv2.HorizontalPodAutoscaler)
+
+	return !reflect.DeepEqual(newObj.Spec, oldObj.Spec)
```

Comment:
I can recall that `reflect.DeepEqual` doesn't always give the correct answer for HPA, please use the approach used here:
https://github.com/kedacore/keda/blob/a2c21ecd30a4e56fcbdbd79c4f04a82d31a83e09/controllers/keda/hpa.go#L162

Reply:
Sure, we can use the same condition.